### PR TITLE
move indicator state from settings.jsx into base component

### PR DESCRIPTION
**Pull request by macqueen**
not sure if this makes things much better, but at least moves the indicator stuff into one place

---

This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry/4151)

### CODE REVIEW

File: src/sentry/static/sentry/app/components/group/issuePluginActions.jsx [hunk 1]
```diff
@@ -77,106 +98,88 @@ const IssuePlugin = React.createClass({
       state.error = {message: t('An unknown error occurred.')};
     }
     this.setState(state);
-  },
+  }
 
   fetchData() {
-    this.setState({
-      loading: true
+    this.onLoad(() => {
```
Comment (dcramer) [inline src/sentry/static/sentry/app/components/group/issuePluginActions.jsx]: will everything use onLoad? If so maybe we just have settings call onLoad automatically (vs wiring up fetchData)

**Review by dcramer — APPROVED** [review]
Generally seems good. I realized last night, when exploring moving the teamwork integration into sentry-plugins, that it will need to be able to override the CreateIssue flow with its own component (for similar reasons as Jira).

File: src/sentry/static/sentry/app/components/group/issuePluginActions.jsx (hunk 1 above)
Comment (macqueen) [inline src/sentry/static/sentry/app/components/group/issuePluginActions.jsx]: added this to the constructor of the base class:
```javascript
if (this.fetchData) {
  this.fetchData = this.onLoad.bind(this, this.fetchData.bind(this));
}
if (this.onSubmit) {
  this.onSubmit = this.onSave.bind(this, this.onSubmit.bind(this));
}
```
so a component that inherits from it and uses `fetchData` and `onSubmit` as method names will just get that state management automatically

**Review by dcramer — APPROVED** [review]
🚀

**Pull request merged**